### PR TITLE
VACMS-9410: Clear event registration info when toggling states.

### PR DESCRIPTION
**Pull request by dsasser**
## Description

Relates to #9410

## QA steps
1) Go to https://pr9593-z2tm41jng73mcitwl40kpzruinndwc2h.ci.cms.va.gov/node/add/event
2) Fill out all required fields
3) Click the box for "Include registration information"
4) Select an option other than "none" for Call to action.
5) Put in a URL
6) Change the Call to Action select to "- None -"
7. +Save the node.
- [x] +Validate that no url is showing on node view. 
8. +Edit the node. and change the CTA to an option other than -none-.
- [x] 7) Confirm that the URL is now blank
8) Select an option other than "none" for Call to action.
9) Put in a URL
10) Uncheck the "Include registration information" checkbox
11) Check the "Include registration information" checkbox
- [x] 12) Confirm that the Call to Action is now set to "- None -" and the URL is empty.

13) +Edit the node.
14) +Set the call to action to "Apply".
15) +Enter a url.
16) +save the node
17) +Edit the node
18) +Uncheck the "Include registration information"
19) +save the node
- [ ]  20) +Validate that the node saves with no values for the url or its type. 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### CODE REVIEW

File: docroot/modules/custom/va_gov_events/js/eventFormHelpers.es6.js [hunk 1]
```diff
@@ -15,12 +15,16 @@
...
     );
+    const ctaUri = document.getElementById("edit-field-link-0-uri") || {};
+    const ctaSelect = document.getElementById("edit-field-event-cta") || {};
     const toggleVal = !!includeRegistrationsBool.checked;
     targetRegistrationElements.forEach((element) => {
       if (toggleVal) {
         element.style.display = "block";
       } else {
         element.style.display = "none";
+        ctaUri.value = "";
```
[the remark this answers is not in the document]
Comment (ryguyk) [reply docroot/modules/custom/va_gov_events/js/eventFormHelpers.es6.js:26]: To me, this feels more how I'd naturally think about what needs to be done:

If the toggleVal is checked, then do two things:
1. Hide elements
2. Blank out some fields

```
if(toggleVal) {
	targetRegistrationElements.forEach((element) => {
	  element.style.display = "block";
	};
  ctaUri.value = "";
  ctaSelect.value = "_none";
} else {
  targetRegistrationElements.forEach((element) => {
	  element.style.display = "none";
	};
}
```

[the remark this answers is not in the document]
Comment (ryguyk) [reply docroot/modules/custom/va_gov_events/js/eventFormHelpers.es6.js:26]: Or, possibly even better:

```
if(toggleVal) {
  toggleRegistrationVisibility(true);
  ctaUri.value = "";
  ctaSelect.value = "_none";
} else {
  toggleRegistrationVisibility(false);
}

const toggleRegistrationVisibility = (show = true) => {
  const displayStyle = show ? 'block' : 'none';
  targetRegistrationElements.forEach((element) => {
	  element.style.display = displayStyle;
	};
}
```

**Review by swirtSJW — CHANGES_REQUESTED** [review]
I added some testing steps because it seemed like a few cases were missing.   A lot of the QA steps identified did not involve saving the various permutations.   I surfaced the last one (steps 13 -20)
image

Where unchecking  the box and then saving fails form validation and the node can not be saved, but no error is visible because of the field that is failing is hidden.

File: docroot/modules/custom/va_gov_events/js/eventFormHelpers.es6.js [hunk 2]
```diff
@@ -15,13 +15,17 @@
...
+      ctaSelect.value = "_none";
+      elementDisplayStyle = 'none';
+    }
     targetRegistrationElements.forEach((element) => {
-      if (toggleVal) {
-        element.style.display = "block";
-      } else {
-        element.style.display = "none";
-      }
+      element.style.display = elementDisplayStyle;
```
Comment (ryguyk) [inline docroot/modules/custom/va_gov_events/js/eventFormHelpers.es6.js:28]: I like this solution. Looks great!

**Review by swirtSJW — CHANGES_REQUESTED** [review]
@dsasser I am seeing another issue related to the "expected behavior: When you change your mind on conditionals, they remove the other field values."   The solution right now is focussed on the CTA, but there is also this other field under the conditional that is not getting cleared out

image

So when I save the node, that registration required is still showing.

image

**Review by swirtSJW — APPROVED** [review]
This all tests out great.  Nice work @dsasser

**Pull request merged**